src/trans_table.rs

```diff
@@ -192,19 +232,19 @@ impl TryFrom<u8> for TranslationTable {
     }
 }
 
-pub fn reverse_complement_bytes(dna: &[u8]) -> Result<Vec<u8>, TranslationError> {
-    let mut v = vec![0u8; dna.len()];
-    for (i, &b) in dna.iter().enumerate() {
-        let n = Nucleotide::try_from(b)?;
-        v[dna.len() - 1 - i] = n.complement().to_ascii();
-    }
-    Ok(v)
+pub fn reverse_complement_bytes<T: NucleotideLike>(
+    dna: &[u8],
+) -> Result<Vec<u8>, TranslationError> {
+    let results = dna
+        .iter()
+        .rev()
+        .map(|b| T::try_from(*b).map(|n| n.complement().to_ascii()));
+
+    // Rust nicely collects an iterator over Result<T, E> into Result<Vec<T>, E> stopping at the first Err:
+    // https://stackoverflow.com/q/63798662/257418
+    results.collect()
```

Comment:
Did you check the perf on this? IIRC I had it the way it was before because it generated better code, but that was several rust versions ago to be fair.

Reply:
Oh, you're right, the perf on this .collect() is worse. Surprisingly, on the other function that doesn't deal with `Result<>` it's better: it looks exactly the same as a manual loop but it avoids an initial call to `memset`. So I'll leave only that one.

Reply:
Huh, weird.